jwst/regtest/test_nircam_image.py

```diff
@@ -0,0 +1,115 @@
+    args = ["config/calwebb_image2.cfg", rtdata.input,
+        "--steps.assign_wcs.save_results=True",
+        ]
+    Step.from_cmdline(args)
+
+    # Grab rest of _rate files for the asn and run image2 pipeline on each to
+    # produce fresh _cal files for the image3 pipeline.  We won't check these
+    # or look at intermediate products
+    rate_files = [
+    "nircam/image/jw42424001001_01101_00001_nrcb5_rate.fits",
+    "nircam/image/jw42424001001_01101_00002_nrca5_rate.fits",
+    "nircam/image/jw42424001001_01101_00002_nrcb5_rate.fits",
+    "nircam/image/jw42424001001_01101_00003_nrca5_rate.fits",
+    "nircam/image/jw42424001001_01101_00003_nrcb5_rate.fits",
+    ]
+    for rate_file in rate_files:
+        rtdata.get_data(rate_file)
+        args = ["config/calwebb_image2.cfg", rtdata.input,
+            "--steps.resample.skip=True"]
+        Step.from_cmdline(args)
+
+    # Get the level3 assocation json file (though not its members) and run
+    # image3 pipeline on all _cal files listed in association
+    rtdata.get_data("nircam/image/jw42424-o002_20191220t214154_image3_001_asn.json")
+    args = ["config/calwebb_image3.cfg", rtdata.input,
+        # Comment out following lines, as the dataset is currently broken
+        # "--steps.tweakreg.save_results=True",
+        # "--steps.skymatch.save_results=True",
+        ]
+    Step.from_cmdline(args)
+
+    return rtdata
+
+
+@pytest.mark.bigdata
+@pytest.mark.parametrize("suffix", ["dq_init", "saturation", "superbias",
+    "refpix", "linearity", "trapsfilled", "dark_current", "jump", "rate",
+    "assign_wcs", "cal", "i2d"])
+def test_nircam_image_stages12(run_pipelines, fitsdiff_default_kwargs, suffix):
+    """Regression test of detector1 and image2 pipelines performed on NIRCam data."""
+    rtdata = run_pipelines
+    rtdata.input = "jw42424001001_01101_00001_nrca5_uncal.fits"
+    output = "jw42424001001_01101_00001_nrca5_" + suffix + ".fits"
+    rtdata.output = output
+    assert os.path.exists(rtdata.output)
+
+    rtdata.get_truth("truth/test_nircam_image_stages/" + output)
+
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+
+@pytest.mark.bigdata
+def test_nircam_image_stage3_i2d(run_pipelines, fitsdiff_default_kwargs):
+    rtdata = run_pipelines
+    rtdata.input = "jw42424-o002_20191220t214154_image3_001_asn.json"
+    rtdata.output = "jw42424-o002_t001_nircam_clear-f444w_i2d.fits"
+    rtdata.get_truth("truth/test_nircam_image_stages/jw42424-o002_t001_nircam_clear-f444w_i2d.fits")
+
+    fitsdiff_default_kwargs['ignore_fields'] = ['date', 'filename']
+    fitsdiff_default_kwargs['ignore_keywords'] += ['naxis1', 'tform*']
+    diff = FITSDiff(rtdata.output, rtdata.truth, **fitsdiff_default_kwargs)
+    assert diff.identical, diff.report()
+
+
+@pytest.mark.bigdata
+def test_nircam_image_stage3_catalog(run_pipelines):
+    rtdata = run_pipelines
```

Comment:
So are all the pipelines actually being run twice: once in "test_nircam_image_stage3_id" and again here in "test_nircam_image_stage3_catalog"? Why can't the comparisons of the 2 different types of outputs (i2d and cat) be combined into a single test module? Neither of them is parameterized, so I would think that the straightforward pieces of code that do the comparisons could be merged together. Or do you want them split up so that they show up as separate failures (in the case where one or both of them fail their assert)?

Reply:
Yes, I want there to be a single file that needs to be OKified if a test fails.  And since the comparisons are not similar, better to break them up into separate tests.

Reply:
But now, the pipeline is just being run once.  Just separate tests that then test aspects of the pipeline.